reset-pass/account-validator-tool/src/java/org/sakaiproject/accountvalidator/tool/controller/MainController.java

```diff
@@ -0,0 +1,1283 @@
+            model.addAttribute("warnMsg", messageSource.getMessage("msg.noSuchValidation", new String[]{va.getValidationToken()}, userLocale));
+            return "error.html";
+
+        } else if (ValidationAccount.STATUS_CONFIRMED.equals(va.getStatus())) {
+
+            model.addAttribute("warnMsg", messageSource.getMessage("msg.alreadyValidated", null, userLocale));
+            return "error.html";
+            
+        } else if (ValidationAccount.STATUS_EXPIRED.equals(va.getStatus())) {
+            
+            model.addAttribute("warnMsg",  messageSource.getMessage("msg.alreadyValidated", new String[]{va.getValidationToken()}, userLocale));
+            return "error.html";
+
+        } else if (sendLegacyLinksEnabled()) {
+
+            return "validate.html";
+        } 
+        
+        User u = null;
+
+        try {  
+                u = userDirectoryService.getUser(va.getUserId());
+        
+        } catch (UserNotDefinedException e){
+            log.error("user ID does not exist for ValidationAccount with tokenId: " + va.getValidationToken());
+            model.addAttribute("warnMsg", messageSource.getMessage("validate.userNotDefined", new String[]{uiService}, userLocale));
+            return "error.html";
+        }
+
+        //user who added this person
+        User addedBy = u.getCreatedBy();
+        
+        //Do not display sites and other welcome page information if user is updating userid
+        if(va.getAccountStatus().equals(ValidationAccount.ACCOUNT_STATUS_USERID_UPDATE)) {
+            String submitUpdate = messageSource.getMessage("submit.update", null, userLocale);
+            model.addAttribute("accountTitle", submitUpdate);
+            model.addAttribute("eid", va.getEid());
+            model.addAttribute("submitBtn", submitUpdate);
+
+        } else {
+            model.addAttribute("accountTitle", messageSource.getMessage("activateAccount.title", null, userLocale));
+            model.addAttribute("eid", u.getDisplayId());
+            model.addAttribute("wait1", messageSource.getMessage("validate.wait.newUser.1", null, userLocale));
+            String linkText = messageSource.getMessage("validate.wait.newUser.2", new String[]{uiService}, userLocale);
+            String transferMembershipsURL = getViewURL("transferMemberships", va);
+            String labelLink = "<a href="+transferMembershipsURL+">"+linkText+"</a>";
+            model.addAttribute("wait2", labelLink);
+            model.addAttribute("addDetailsSub", messageSource.getMessage("submit.new.account", null, userLocale));
+            model.addAttribute("submitBtn", messageSource.getMessage("submit.new.account", null, userLocale));
+            //we need to know which sites they're a member of:
+            Set<String> groups = authzGroupService.getAuthzGroupsIsAllowed(EntityReference.getIdFromRef(va.getUserId()), "site.visit", null);
+            Iterator<String> git = groups.iterator();
```

Comment:
I don't think we need an iterator here, you can loop a set:

for (String groupRef : groups)